src/read.rs

```diff
@@ -59,9 +60,19 @@ where
 {
     // tell writer about our epoch tracker
     let epoch = sync::Arc::new(atomic::AtomicUsize::new(0));
-    inner.epochs.lock().unwrap().push(Arc::clone(&epoch));
+
+    {
```

Comment:
This lock is taken so rarely that I don't know that it's worthwhile to turn it into a `parking_lot::Mutex`. The writer does take it on every `refresh`, but it should be entirely uncontended, and pretty much irrelevant. It'd be good to see a benchmark of this to see that it actually makes a difference before adding this complexity.

Reply:
Honestly, I still feel more comfortable with `parking_lot` if possible, given its additional safety features, such as no lock poisoning.

Reply:
Hmm, I'm not sure I consider no lock poisoning a safety feature? To the contrary, it means that you don't find out that the previous holder of the lock crashes, and thus may have corrupted the locked value's contents.

Reply:
Almost all panics (aside from OOM errors) would be from user code, right? In that case, they would have to panic before an operation is added to the oplog or before a call to `refresh`. The oplog itself and all other members of the writer are uncorrupted, so the lock can be safely unlocked via RAII.

That would be an issue if they were adding many values in a loop, but only logically for their program, and they should probably handle that whenever their worker thread crashes, as it did not get a chance to finish adding all values. I mean, we can really only guarantee everything works up until the point it crashes. Guaranteeing eventual consistency still holds so long as their program can restart the worker thread and presumably *not* crash when adding the values again.

Reply:
Actually, I just realized, the lock here is never even used in conjunction with user code, so it has zero chance of being poisoned anyway. I'm not sure if that's a point in favor or against keeping `parking_lot`.

Reply:
I think that assumes that users aren't doing things that can panic in `Clone` for example. We clone in `refresh`, so if that panics for whatever reason, then later cloning a read handle wouldn't crash if you don't track poison. I think benchmarking this first is a good idea -- if the performance gains are negligible (which I think they'll be), then I think we should stick with the `std` lock.